Code/Tools/FBuild/FBuildWorker/Worker/Worker.cpp

```diff
@@ -268,8 +268,16 @@ void Worker::UpdateAvailability()
     }
 
     WorkerThreadRemote::SetNumCPUsToUse( numCPUsToUse );
-
-    m_WorkerBrokerage.SetAvailability( numCPUsToUse > 0);
+    
+    if ( numCPUsToUse > 0 )
+    {
+        m_WorkerBrokerage.SetAvailable();
+    }
+    else
+    {
+        // to avoid doing a lot of network file I/O when going idle and not idle,
+        // don't set unavailable here
```

Comment:
It now looks like a worker will never remove its token (until shutdown). This is undesirable - removing the token prevents other workers connecting to a worker that won't do any work for it (and using up a connection slot).